### PR TITLE
Change error message and code for conflicts, unless if-match was used

### PR DESCRIPTION
**Pull request by graetzer**
Use 409 (Conflict) error code i

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11146/

### CODE REVIEW

File: arangod/RestHandler/RestGraphHandler.cpp [hunk 1]
```diff
@@ -816,7 +810,9 @@ Result RestGraphHandler::documentModify(graph::Graph& graph, const std::string&
   }
 
   if (result.fail()) {
-    generateTransactionError(result);
+    // simon: do not pass in collection name, otherwise HTTP return code
+    //        changes to 404 in for unkown _to/_from collection -> breaks API
```
Comment (jsteemann) [inline arangod/RestHandler/RestGraphHandler.cpp:814]:
```suggestion
    //        changes to 404 in for unknown _to/_from collection -> breaks API
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr/11160/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11164/

**Comment by jsteemann** [comment]
Somehow this PR breaks the v8-users catch tests.
Also some other tests are failing:
```
* Test "http_server"
    [FAILED]  tests/rb/HttpInterface/api-document-delete-spec.rb

      "createADocumentAndDeleteIt,UsingIf-Match" failed: ArangoDB delete a document: deleting documents: create a document and delete it, using if-match
id: './tests/rb/HttpInterface/api-document-delete-spec.rb[1:1:2:2]'
description: 'create a document and delete it, using if-match'
full_description: >-
  ArangoDB delete a document: deleting documents: create a document and delete
  it, using if-match
status: failed
file_path: ./tests/rb/HttpInterface/api-document-delete-spec.rb
line_number: 126
run_time: 0.051738749
pending_message: null
exception:
  class: 'RSpec::Expectations::ExpectationNotMetError'
  message: |

    expected: "_a1BOdJG---"
         got: "386897"

    (compared using ==)

* Test "ssl_server"
    [FAILED]  tests/rb/HttpInterface/api-document-delete-spec.rb

      "createADocumentAndDeleteIt,UsingIf-Match" failed: ArangoDB delete a document: deleting documents: create a document and delete it, using if-match
id: './tests/rb/HttpInterface/api-document-delete-spec.rb[1:1:2:2]'
description: 'create a document and delete it, using if-match'
full_description: >-
  ArangoDB delete a document: deleting documents: create a document and delete
  it, using if-match
status: failed
file_path: ./tests/rb/HttpInterface/api-document-delete-spec.rb
line_number: 126
run_time: 0.059320837
pending_message: null
exception:
  class: 'RSpec::Expectations::ExpectationNotMetError'
  message: |

    expected: "_a1BLCqe---"
         got: "386897"

    (compared using ==)
```

**Comment by graetzer** [comment]
can you please refer to the jenkins run with the errors above. I just see the catch issue

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11164/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/job/arangodb-matrix-pr-linux/12848/EDITION=community,STORAGE_ENGINE=rocksdb,TEST_SUITE=cluster,limit=(linux&&test)%7C%7Cgce/artifact/testfailures.txt

**Comment by graetzer** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11182/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/11190/

File: tests/Mocks/StorageEngineMock.cpp [hunk 2]
```diff
@@ -1353,9 +1353,8 @@ arangodb::Result PhysicalCollectionMock::updateInternal(
       }
       TRI_ASSERT(doc.isObject());
       TRI_voc_rid_t oldRev = TRI_ExtractRevisionId(doc);
-      int res = checkRevision(trx, expectedRev, oldRev);
-      if (res != TRI_ERROR_NO_ERROR) {
-        return arangodb::Result(res);
+      if (!checkRevision(trx, expectedRev, oldRev)) {
+        return arangodb::Result(TRI_ERROR_ARANGO_CONFLICT, " _rev values mismatch");
```
Comment (jsteemann) [inline tests/Mocks/StorageEngineMock.cpp:1357]:
```suggestion
        return arangodb::Result(TRI_ERROR_ARANGO_CONFLICT, "_rev values mismatch");
```

**Pull request merged**